rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -87,11 +87,9 @@ NodeGraph::get_topic_names_and_types(bool no_demangle) const
 
   ret = rcl_names_and_types_fini(&topic_names_and_types);
   if (ret != RCL_RET_OK) {
-    auto ex_msg = rcl_get_error_string().str;
-    rcl_reset_error();
     // *INDENT-OFF*
     throw std::runtime_error(
-      std::string("could not destroy topic names and types: ") + ex_msg);
+      std::string("could not destroy topic names and types: ") + rcl_get_error_string().str);
```

Comment:
Why rcl_reset_error() is removed ?

`RCL_CHECK_ARGUMENT_FOR_NULL(topic_names_and_types, RCL_RET_INVALID_ARGUMENT);`  is called in rcl_names_and_types_fini().

Reply:
because we cannot do that unless copy the message buffer on somewhere else. once `rcl_reset_error()` is called, the global message buffer will be deleted immediately.

Reply:
The rcl_get_error_string().str has already retrieved the rcl error log and saved it to the exception message.    
So I think we can clear the rcl error log.

Reply:
originally i thought that too, but this fails to the test cases since garbled characters in the exception message. so i removed this change from this PR.